chips/nrf52/src/ble_radio.rs

```diff
@@ -592,9 +588,21 @@ impl<'a> Radio<'a> {
         self.registers.txpower.set(self.tx_power.get() as u32);
     }
 
-    fn set_dma_ptr(&self) {
+    fn replace_dma_ptr(&self, buf: &'static mut [u8]) -> &'static mut [u8] {
         unsafe {
-            self.registers.packetptr.set(PAYLOAD.as_ptr() as u32);
+            // Get buffer encoded in in DMA registers. If it's a null
+            // pointer, the length should also be zero. Otherwise, use
+            // the configured length as the old length.
+	    // This should be the only place these registers are modified or read.
+            let old_base = self.registers.packetptr.get() as usize as *mut u8;
+	    let old_len = if old_base.is_null() {
+		0
+	    } else {
+		self.registers.pcnf1.read(PacketConfiguration1::MAXLEN) as usize
+	    };
+            self.registers.packetptr.set(buf.as_ptr() as u32);
+	    self.registers.pcnf1.modify(PacketConfiguration1::MAXLEN.val(buf.len() as u32));
+	    slice::from_raw_parts_mut(old_base, old_len)
         }
```

Comment:
It looks like `old_base` can be null at the `slice::from_raw_parts_mut` invocation, which is UB.

Please add a safety comment for `slice::from_raw_parts_mut`, and minimize the size of this `unsafe` block (is `slice::from_raw_parts_mut` the only unsafe operation here?).